src/perl/lib/wtsi_clarity/epp/sm/pico_analyser.pm

```diff
@@ -59,8 +66,9 @@ override 'run' => sub {
   my $pdf = wtsi_clarity::util::pdf::factory::pico_analysis_results->new()->build($parameters);
 
   # Attach PDF to process
-  my $date = DateTime->now()->strftime('%Y-%m-%d_%H-%M-%S');
-  $pdf->saveas(qq{./$stock_plate.$date.pdf});
+  my $date = DateTime->now()->strftime('%Y%m%d%H%M%S');
+  my $filename = sprintf $FILE_NAME, $self->output_file_limsid, $stock_plate, $date;
```

Comment:
If you make a method of this, then it became testable, whether we got the right file name or not.